### PR TITLE
Add zipfile decorator

### PR DESCRIPTION
**Pull request by binary1230**
Huh. turns out this was actually really easy to do. part of #1397 

I wanted to the ability to take a bunch of our existing CSV reports and stick them into a .zip file.  If you stick the ```@multifile_zipfile``` decorator on a function, it'll pass you a ```ZipFile``` object which you can use to easily create a bunch of files inside a zipfile.

result from the test data in here:
image

I'll be updating the stuff in summary.py to have real data in it in the next PR.

### CODE REVIEW

File: uber/site_sections/summary.py [hunk 1]
```diff
@@ -110,6 +110,14 @@ def personalized_badges(self, out, session):
                                                 Attendee.amount_extra >= c.SUPPORTER_LEVEL).order_by(Attendee.full_name).all():
             out.writerow(['', 'Supporter', a.badge_printed_name or a.full_name])
```
Comment (EliAndrewC) [inline uber/site_sections/summary.py]: This is indeed a good example of the kind of thing you can do, but I assume we're not actually merging this in with these test files and such, right?

File: uber/decorators.py [hunk 2]
```diff
@@ -87,6 +87,22 @@ def returns_json(*args, **kwargs):
     return returns_json
```
Comment (EliAndrewC) [inline uber/decorators.py]: +1 on this decorator, looks great!

File: uber/common.py [hunk 3]
```diff
@@ -16,7 +17,7 @@
 import traceback
 from glob import glob
 from uuid import uuid4
-from io import StringIO
+from io import StringIO, BytesIO
```
Comment (EliAndrewC) [inline uber/common.py]: Can you move this down to be consistent with our ordering?  These imports are all in increasing order of line length.  I find it very soothing to look at.  (That sounds like a joke, but I genuinely like how it looks to see the import lines getting monatomically longer.

Comment (binary1230) [inline uber/common.py]: kkkkkkkkkkkkkkkkkkkkkkkkkkkkkkkkkkkkkkkkkkkkkkkk

for @EliAndrewC , anything.

File: uber/site_sections/summary.py (hunk 1 above)
Comment (binary1230) [inline uber/site_sections/summary.py]: I was going to do another PR on top of this one that replaces this with the stuff from #1397, and not merge this one in until then.  mostly just wanted your eyes on the decorator so I wasn't going down the wrong path first.

Comment (EliAndrewC) [inline uber/site_sections/summary.py]: Gotcha.  Yeah, the decorator looks great, definitely how we want to be doing this kind of thing.

**Comment by kitsuta** [comment]
+1
